src/components/app-single-post/test/app-single-post.spec.tsx

```diff
@@ -0,0 +1,50 @@
+import { h } from '@stencil/core';
+import { newSpecPage } from '@stencil/core/testing';
+import { BlogPostService } from '../../../services/blogPost';
+import { AppSinglePost } from '../app-single-post';
+
+describe('app-single-post', () => {
+  beforeEach(() => {
+    BlogPostService.getBlogPost = jest.fn().mockResolvedValue({});
+  });
+  it('should call delete method of blogPost service when delete button is clicked', async () => {
+    //arrange
+    const page = await newSpecPage({
+      components: [AppSinglePost],
+      template: () => (
+        <app-single-post
+          //@ts-ignore
+          match={{ params: { id: '1' } }}
+          //@ts-ignore
+          history={{ goBack() {} }}
+        ></app-single-post>
+      ),
+    });
+    BlogPostService.deleteBlogPost = jest.fn();
+    await page.waitForChanges();
+    //act
+    page.rootInstance.handleDelete();
+    //assert
+    expect(BlogPostService.deleteBlogPost).toHaveBeenCalledTimes(1);
+  });
+  it('should call update method og blogPost service when makes changes button is clicked', async () => {
+    //arrange
+    const page = await newSpecPage({
```

Comment:
this is repeated in above test, so can be put in beforeEach, always avoid duplicate code even if its a test